tests/Feature/CountryControllerTest.php

```diff
@@ -0,0 +1,189 @@
+    {
+        yield "country with empty credentials is invalid" => [
+            "data" => [
+                "name" => null,
+                "longitude" => null,
+                "latitude" => null,
+                "iso" => null,
+            ],
+            "expectedErrors" => ["name", "latitude", "longitude", "iso"],
+        ];
+
+        yield "country with incorrect name is invalid" => [
+            "data" => [
+                "name" => "poland",
+                "longitude" => 21.555,
+                "latitude" => 55.234,
+                "iso" => "pl",
+            ],
+            "expectedErrors" => ["name"],
+        ];
+
+        yield "country with incorrect longitude is invalid" => [
+            "data" => [
+                "name" => "Poland",
+                "longitude" => "21string",
+                "latitude" => 55.234,
+                "iso" => "pl",
+            ],
+            "expectedErrors" => ["longitude"],
+        ];
+
+        yield "country with incorrect latitude is invalid" => [
+            "data" => [
+                "name" => "Poland",
+                "longitude" => 21.555,
+                "latitude" => "55.234string",
+                "iso" => "pl",
+            ],
+            "expectedErrors" => ["latitude"],
+        ];
+
+        yield "country with incorrect iso is invalid" => [
+            "data" => [
+                "name" => "Poland",
+                "longitude" => 21.555,
+                "latitude" => 55.234,
+                "iso" => "Pl",
+            ],
+            "expectedErrors" => ["iso"],
+        ];
+    }
+
+    public function testCountryUpdate(): void
+    {
+        $data = [
+            "name" => "Poland",
+            "latitude" => 44.543,
+            "longitude" => -43.122,
+            "iso" => "pl",
+        ];
+
+        $country = Country::factory()->create();
+
+        $this->patch(route(name:"countries.update", parameters: ["country" => $country]), $data);
+
+        $this->assertDatabaseHas(table:"countries", data: $data);
+    }
+
+    public function testCountryUpdateIsRefusingDuplicateName(): void
+    {
+        $country = Country::query()->create([
+            "name" => "Romania",
+            "latitude" => -55.54323,
+            "longitude" => 42.3721,
+            "iso" => "rom",
+        ]);
+
+        $countryToCompare = Country::query()->create([
+            "name" => "Poland",
+            "latitude" => 44.543,
+            "longitude" => -43.122,
+            "iso" => "pl",
+        ]);
+
+        $updateCountryToCompare = [
+            "name" => $countryToCompare["name"],
+            "latitude" => $countryToCompare["latitude"],
+            "longitude" => $countryToCompare["longitude"],
+            "iso" => $country["iso"],
+        ];
+
+        $response = $this->patch(route(name:"countries.update", parameters: ["country" => $countryToCompare]), data: $updateCountryToCompare);
+        $response->assertSessionHasErrors(["iso"]);
+    }
```

Comment:
Why so complicated?
```php
Country::factory()->create([
           "name" => "Poland"
            "iso" => "pl",
        ]);

$countryToUpdate = Country::factory()->create([
           "name" => "Romania"
            "iso" => "rom",
        ]);

        $this->patch(route(name:"countries.update", parameters: ["country" => $countryToUpdate]), data: [
       "name" => "Romania,
       "latitude" => 1,
       "longtitude" => 2,
       "iso" => "pl",
        ])
        ->assertSessionHasErrors(["iso"]);
```

Also test name is misleading.
There you are testing that ISO cannot be updated if already exists in database for another country.

Reply:
Yes, i didn't notice test name mismatch.